features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SystemRequirementConfiguration.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import android.os.Handler
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.core.internal.utils.submitSafe
+import com.datadog.android.sessionreplay.internal.prerequisite.CPURequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.MemoryRequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.SystemRequirementChecker
+
+/**
+ * This class defines the system requirements necessary to enable the Session Replay feature.
+ */
+class SystemRequirementConfiguration internal constructor(
+    internal val minCPUCores: Int,
+    internal val minRAMSizeMb: Int
+) {
+
+    internal fun runIfMeetRequirements(
+        sdkCore: FeatureSdkCore,
+        uiHandler: Handler,
+        internalLogger: InternalLogger,
+        runnable: () -> Unit
+    ) {
+        val checkers = listOf(
+            CPURequirementChecker(minCPUCores, internalLogger = internalLogger),
+            MemoryRequirementChecker(minRAMSizeMb, internalLogger = internalLogger)
+        )
+        (sdkCore as InternalSdkCore).getPersistenceExecutorService().submitSafe(OPERATION_NAME, internalLogger) {
+            val checkResult = checkers.all {
+                it.checkMinimumRequirement()
+            }
```

Comment:
> [!WARNING]
> So I have a bit of a concern here. Right now it's not a problem, but it could become one eventually. 
> We create a new Checker instance and perform the checks every time we call the `runIfMeetRequirements()` method. The thing is that the cpu and ram size will never change on a given device, meaning we could memoize the result of the check for the current session and ensure we don't perform the check on every call. Not necessary for this PR but we should probably add a task for this.

Reply:
I see the point here, so the way of memorising it for me is to persist result in shared preference or a file, but I may wonder if this solution will have the same effect with the current one because current checkers actually just read files. So the difference of two solutions is just the file that they are reading if I am not wrong?